===================================================================
RCS file: /cvsroot/firehol/firehol/firehol.sh,v
retrieving revision 1.224
retrieving revision 1.226
diff -u -r1.224 -r1.226
--- firehol/firehol/firehol.sh	2005/01/24 22:24:48	1.224
+++ firehol/firehol/firehol.sh	2005/01/25 21:28:19	1.226
@@ -144,7 +144,7 @@
 # Directories and files
 
 # These files will be created and deleted during our run.
-FIREHOL_DIR="/tmp/firehol-tmp-$$"
+FIREHOL_DIR="/tmp/.firehol-tmp-$$-${RANDOM}-${RANDOM}"
 FIREHOL_CHAINS_DIR="${FIREHOL_DIR}/chains"
 FIREHOL_OUTPUT="${FIREHOL_DIR}/firehol-out.sh"
 FIREHOL_SAVED="${FIREHOL_DIR}/firehol-save.sh"
@@ -211,7 +211,7 @@
 # Externally defined services can be placed in "${FIREHOL_CONFIG_DIR}/services/"
 if [ ! -d "${FIREHOL_CONFIG_DIR}/services" ]
 then
-	"${MKDIR_CMD}" -p "${FIREHOL_CONFIG_DIR}/services"
+	"${MKDIR_CMD}" "${FIREHOL_CONFIG_DIR}/services"
 	if [ $? -ne 0 ]
 	then
 		echo >&2
@@ -239,14 +239,14 @@
 		exit 1
 	fi
 fi
-"${MKDIR_CMD}" -p "${FIREHOL_DIR}"				|| exit 1
-"${MKDIR_CMD}" -p "${FIREHOL_CHAINS_DIR}"			|| exit 1
+"${MKDIR_CMD}" "${FIREHOL_DIR}"				|| exit 1
+"${MKDIR_CMD}" "${FIREHOL_CHAINS_DIR}"			|| exit 1
 
 
 # Make sure we have a directory for our data.
 if [ ! -d "${FIREHOL_SPOOL_DIR}" ]
 then
-	"${MKDIR_CMD}" -p "${FIREHOL_SPOOL_DIR}"		|| exit 1
+	"${MKDIR_CMD}" "${FIREHOL_SPOOL_DIR}"			|| exit 1
 	"${CHOWN_CMD}" root:root "${FIREHOL_CONFIG_DIR}"	|| exit 1
 	"${CHMOD_CMD}" 700 "${FIREHOL_CONFIG_DIR}"		|| exit 1
 fi
@@ -1055,7 +1055,7 @@
 	local x=
 	for x in ${servers}
 	do
-		local tmp="${FIREHOL_DIR}/firehol.rpcinfo.$$"
+		local tmp="${FIREHOL_DIR}/firehol.rpcinfo.$$.${RANDOM}"
 		
 		set_work_function "Getting RPC information from server '${x}'"
 		
@@ -1169,7 +1169,7 @@
 	local x=
 	for x in ${servers}
 	do
-		local tmp="${FIREHOL_DIR}/firehol.rpcinfo.$$"
+		local tmp="${FIREHOL_DIR}/firehol.rpcinfo.$$.${RANDOM}"
 		
 		set_work_function "Getting RPC information from server '${x}'"
 		
@@ -2567,9 +2567,9 @@
 if [ -f "/proc/config" ]
 then
 	KERNEL_CONFIG="/proc/config"
-	${CAT_CMD} /proc/config >/tmp/kcfg.$$
-	source /tmp/kcfg.$$
-	${RM_CMD} -f /tmp/kcfg.$$
+	${CAT_CMD} /proc/config >${FIREHOL_DIR}/kcfg
+	source ${FIREHOL_DIR}/kcfg
+	${RM_CMD} -f ${FIREHOL_DIR}/kcfg
 	
 elif [ -f "/lib/modules/`${UNAME_CMD} -r`/build/.config" ]
 then
